src/engine/ReluConstraint.cpp

```diff
@@ -222,33 +252,75 @@ void ReluConstraint::notifyUpperBound( unsigned variable, double newBound )
 
         if ( isActive() )
         {
+            createTighteningRow();
```

Comment:
So, I noticed we have a bunch of "createTighteningRow" calls that are not guarded by an if ( proofs ) condition. I see that in this case they will return without doing nothing, but in order to emphasize that this is just for proofs, I prefer that the guard appear outside, and not inside, the call.